src/kzg10/fastpoly.rs

```diff
@@ -0,0 +1,372 @@
+use crate::Error;
+use ark_ec::PairingEngine;
+use ark_ff::{Field, One, Zero};
+use ark_poly::UVPolynomial;
+
+use ark_poly::polynomial::univariate::DensePolynomial;
+
+type P<E: PairingEngine> = DensePolynomial<E::Fr>;
```

Comment:
Maybe just import this is as `ark_poly::polynomial::univariate::DensePolynomial as Poly`? Also, not sure if it's more ergonomic to make it generic over a pairing engine, instead of directly over the scalar field.

Reply:
That sounds good, and also it may be better to merge the polynomial operations into `algebra` rather than `poly-commit`

Reply:
Yeah, that seems reasonable!

Reply:
Let's merge the polynomial operations here for the time being, and then in a follow up PR we can move them to `ark-poly`?